### PR TITLE
[CI] MacOS stages for branches/tags and PRs when certain conditions

### PR DESCRIPTION
**Pull request by v1v**
## What

- Run the MacOS stages for all the branches/tags builds.
- Run the MacOS stages for PRs that 
  - were built manually with the UI parameter `macosTest`
  - got a GitHub comment with `/test macos`
  - got a GitHub label `macOS`

## Why is it important?

As long as MacOS workers are not fungible/ephemerals we got a restriction in the number of physical workers we got for such we need to disable the macOS by default for all the PRs.

## Issues

Closes https://github.com/elastic/beats/issues/19754

## Tasks

- [x] Test
- [x] Label based

## Tests

- Comment based

<details><summary>Expand to view</summary>
<p>

image

</p>
</details>

- UI params based

<details><summary>Expand to view</summary>
<p>

image

caused

image

</p>
</details>

- GH label based

<details><summary>Expand to view</summary>
<p>

image

and a push event caused 

image

</p>
</details>

## Tasks

- [ ] Agree in https://github.com/elastic/beats/issues/19754

### CODE REVIEW

**Comment by v1v** [comment]
jenkins run the tests for macos please

File: Jenkinsfile [hunk 1]
```diff
@@ -50,17 +50,15 @@ pipeline {
     rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
   }
   triggers {
-    issueCommentTrigger('(?i).*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*')
+    issueCommentTrigger('(?i).*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests\\W+(?:for\\W+macos)?(?:\\W+please)?.*')
```
Comment (kuisathaverat) [inline Jenkinsfile:53]: Could we simplify the message to `/test` to run the job, and `/test macos` or `/test macos windows` to run another tests

```suggestion
    issueCommentTrigger('(?i)(.*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*|^\/test(\W+macos)?(\W+windows)?$)')
```

Comment (kuisathaverat) [reply Jenkinsfile:53]: even do we keep the `jenkins run the test please` message

Comment (kuisathaverat) [reply Jenkinsfile:53]: shit I trigger a build with that, because of that I don't like it.

Comment (v1v) [reply Jenkinsfile:53]: I'll update the README.md accordingly then and commit all together

Comment (v1v) [reply Jenkinsfile]: The comment `/test windows` is not supported at the moment

File: Jenkinsfile [hunk 2]
```diff
@@ -50,17 +50,15 @@ pipeline {
     rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
   }
   triggers {
-    issueCommentTrigger('(?i).*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*')
+    issueCommentTrigger('(?i)(.*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*|^\/test(\W+macos)?$)')
```
Comment (v1v) [inline Jenkinsfile:53]:
```suggestion
    issueCommentTrigger('(?i)(.*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*|^/test(\W+macos)?$)')
```

File: Jenkinsfile [hunk 3]
```diff
@@ -50,17 +50,15 @@ pipeline {
     rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
   }
   triggers {
-    issueCommentTrigger('(?i).*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*')
+    issueCommentTrigger('(?i)(.*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*|^/test(\W+macos)?$)')
```
Comment (v1v) [inline Jenkinsfile:53]:
```suggestion
    issueCommentTrigger('(?i)(.*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*|^/test(\\W+macos)?$)')
```

**Pull request merged**